packages/consensus/src/types/state.cairo

```diff
@@ -0,0 +1,71 @@
+//! State is a top level struct containing the chain state and the utxo set
+
+use crate::types::utreexo::{UtreexoState, UtreexoStateDefault};
+use crate::types::utxo_set::UtxoSet;
+use crate::validation::{
+    difficulty::{validate_bits, adjust_difficulty}, coinbase::validate_coinbase,
+    timestamp::{validate_timestamp, next_prev_timestamps},
+    work::{validate_proof_of_work, compute_total_work}, block::{compute_and_validate_tx_data},
+};
+use super::block::{BlockHash, Block, TransactionData};
+use super::chain_state::ChainState;
+
+#[derive(Default, Drop, Copy, Debug, Serde)]
+pub struct State {
+    pub chain_state: ChainState,
+    pub utreexo_state: UtreexoState
+}
+
+/// Full block validator (w/o bitcoin script checks and utxo inclusion verification for now).
+#[generate_trait]
+pub impl BlockValidatorImpl of BlockValidator {
+    fn validate_and_apply(self: State, block: Block) -> Result<State, ByteArray> {
+        let mut utxo_set = UtxoSet {
```

Comment:
instantiate `utxo_set` using `self. utreexo_state` and a new default cache

Reply:
I can stick to the scope of the issue and instantiate a dummy utreexo but i felt i would be better to modify the whole flow already and use the global state